src/snapshots.rs

```diff
@@ -37,18 +37,28 @@ pub fn capture(path: &Path) -> Result<Option<CaptureStatus>, Error> {
     }
 
     let branch_name = format!("dura/{}", head.id());
-    let branch_commit = find_head(&repo, branch_name.as_str());
-
-    if repo.find_branch(&branch_name, BranchType::Local).is_err() {
-        repo.branch(branch_name.as_str(), &head, false)?;
-    }
+    let branch_commit = match repo.find_branch(&branch_name, BranchType::Local) {
+        Ok(mut branch) => {
+            match branch.get().peel_to_commit() {
+                Ok(commit) if commit.id() != head.id() => Some(commit),
+                _ => {
+                    // Dura branch exist but no commit is made by dura
+                    // So we clean this branch
+                    branch.delete()?;
+                    None
+                }
+            }
+        }
+        Err(_) => None,
+    };
+    let parent_commit = branch_commit.as_ref().unwrap_or(&head);
 
     // tree
     let mut index = repo.index()?;
     index.add_all(["*"].iter(), IndexAddOption::DEFAULT, None)?;
 
     let dirty_diff = repo.diff_tree_to_index(
-        Some(&branch_commit.as_ref().unwrap_or(&head).tree()?),
+        Some(&parent_commit.tree()?),
         Some(&index),
         Some(DiffOptions::new().include_untracked(true)),
     )?;
```

Comment:
Actually had a hard time creating a test for this, we checked for `git status` on https://github.com/tkellogg/dura/blob/a0983336185853f683c589da67226ee772b176f7/src/snapshots.rs#L34-L37, and not sure when the diff delta >= 0 and we dont want to capture :thinking: